### PR TITLE
feat(pipeline): support register lifecycle hook client

### PR DESCRIPTION
**Pull request by chengjoey**
#### What this PR does / why we need it:
1. support register lifecycle hook client
2. refactor lifecycle hook client to pb

#### Which issue(s) this PR fixes:

- Fixes #your-issue_number
- [Erda Cloud Issue Link](https://erda.cloud/erda/dop/projects/387/issues/gantt?filter__urlQuery=eyJhc3NpZ25lZSI6WyIxMDAxMjA1Il19&id=313314&iterationID=1218&pId=0&type=TASK)

#### Specified Reviewers:

/assign @sfwn 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |    support register lifecycle hook client          |
| 🇨🇳 中文    |    lifecycle hook client pb重构 支持注册client          |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: .erda/migrations/pipeline/20220518-pipeline-lifecycle-client-add-uniquekey.sql [hunk 1]
```diff
@@ -0,0 +1,5 @@
+ALTER TABLE `dice_pipeline_lifecycle_hook_clients` MODIFY COLUMN `id` bigint(20) unsigned NOT NULL AUTO_INCREMENT COMMENT '主键';
+
+ALTER TABLE `dice_pipeline_lifecycle_hook_clients` MODIFY COLUMN `name` varchar(128);
+
+ALTER TABLE `dice_pipeline_lifecycle_hook_clients` ADD UNIQUE KEY `uk_name` (`name`);
```
Comment (sfwn) [inline .erda/migrations/pipeline/20220518-pipeline-lifecycle-client-add-uniquekey.sql:5]: unique key or unique index

File: modules/pipeline/endpoints/v2_test.go [hunk 2]
```diff
@@ -136,6 +136,7 @@ func newCreateV2Work(workerID int) (count int) {
 	return
 }
 
+
```
Comment (sfwn) [inline modules/pipeline/endpoints/v2_test.go:139]: useless

File: modules/pipeline/providers/lifecycle_hook_client/life.cycle.service.go [hunk 3]
```diff
@@ -12,34 +12,87 @@
...
+	sync.Mutex
+	hookClientMap map[string]*pb.LifeCycleClient
+	logger        logs.Logger
+	dbClient      *dbclient.Client
+}
 
-func PostLifecycleHookHttpClient(source string, req interface{}, resp interface{}) error {
+func (s *LifeCycleService) LifeCycleRegister(ctx context.Context, req *pb.LifeCycleClientRegisterRequest) (*pb.LifeCycleClientRegisterResponse, error) {
+	if req.Name == "" {
+		return nil, status.Errorf(codes.InvalidArgument, "client name is required")
```
Comment (sfwn) [inline modules/pipeline/providers/lifecycle_hook_client/life.cycle.service.go:45]: which is better compare to apierrors?

**Review by sfwn — CHANGES_REQUESTED** [review]

File: .erda/migrations/pipeline/20220518-pipeline-lifecycle-client-add-uniquekey.sql (hunk 1 above)
Comment (chengjoey) [reply .erda/migrations/pipeline/20220518-pipeline-lifecycle-client-add-uniquekey.sql]: adjusted

File: modules/pipeline/endpoints/v2_test.go (hunk 2 above)
Comment (chengjoey) [reply modules/pipeline/endpoints/v2_test.go]: adjusted

File: modules/pipeline/providers/lifecycle_hook_client/life.cycle.service.go (hunk 3 above)
Comment (chengjoey) [reply modules/pipeline/providers/lifecycle_hook_client/life.cycle.service.go]: adjusted, use `apierrors`

**Review by sfwn — APPROVED** [review]

**Pull request merged**